pkg/inspector/image-inspector.go

```diff
@@ -210,27 +226,30 @@ func (i *defaultImageInspector) pullImage(client *docker.Client) error {
 		return authCfgErr
 	}
 
-	reader, writer := io.Pipe()
-	// handle closing the reader/writer in the method that creates them
-	defer writer.Close()
-	defer reader.Close()
-	imagePullOption := docker.PullImageOptions{
-		Repository:    i.opts.Image,
-		OutputStream:  writer,
-		RawJSONStream: true,
-	}
-
-	bytesChan := make(chan int)
-	go aggregateBytesAndReport(bytesChan)
-	go decodeDockerPullMessages(bytesChan, reader)
-
 	// Try all the possible auth's from the config file
 	var authErr error
 	for name, auth := range imagePullAuths.Configs {
-		if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
-			return nil
+		reader, writer := io.Pipe()
+		// handle closing the reader/writer in the method that creates them
+		defer writer.Close()
+		defer reader.Close()
+		imagePullOption := docker.PullImageOptions{
+			Repository:    i.opts.Image,
+			OutputStream:  writer,
+			RawJSONStream: true,
+		}
+
+		parsedErrors := make(chan error, 1)
+		defer func() { close(parsedErrors) }()
+		go decodeDockerResponse(parsedErrors, reader)
+
+		if err := client.PullImage(imagePullOption, auth); err != nil {
+			return err
```

Comment:
This was authErr that I now changed to just `err` because authentication errors are only returned in the stream. Errors here should terminate the loop in my opinion because they are more general ones that will happen with any authentication.